encrypt-config/ConfigFile.cpp

```diff
@@ -0,0 +1,195 @@
+  if (it != config_lines_.end()) {
+    it->updateValue(value);
+  } else {
+    throw std::invalid_argument{"Key " + key + " not found in the config file!"};
+  }
+}
+
+void ConfigFile::insertAfter(const std::string& after_key, const std::string& key, const std::string& value) {
+  auto it = findKey(after_key);
+  if (it != config_lines_.end()) {
+    ++it;
+    config_lines_.emplace(it, key, value);
+  } else {
+    throw std::invalid_argument{"Key " + after_key + " not found in the config file!"};
+  }
+}
+
+void ConfigFile::append(const std::string& key, const std::string& value) {
+  config_lines_.emplace_back(key, value);
+}
+
+int ConfigFile::erase(const std::string& key) {
+  auto has_this_key = [&key](const ConfigLine& line) { return line.key_ == key; };
+  auto new_end = std::remove_if(config_lines_.begin(), config_lines_.end(), has_this_key);
+  auto num_removed = std::distance(new_end, config_lines_.end());
+  config_lines_.erase(new_end, config_lines_.end());
+  return gsl::narrow<int>(num_removed);
+}
+
+void ConfigFile::writeTo(const std::string& file_path) const {
+  std::ofstream file{file_path};
+  for (const auto& config_line : config_lines_) {
+    file << config_line.line_ << '\n';
+  }
+}
+
+int ConfigFile::encryptSensitiveProperties(const utils::crypto::Bytes& encryption_key) {
+  int num_properties_encrypted = 0;
+
+  for (const auto& property_key : getSensitiveProperties()) {
+    utils::optional<std::string> property_value = getValue(property_key);
+    if (!property_value) { continue; }
+
+    std::string encryption_type_key = property_key + ".protected";
+    utils::optional<std::string> encryption_type = getValue(encryption_type_key);
+
+    if (!encryption_type || encryption_type->empty() || *encryption_type == "plaintext") {
+      std::string encrypted_property_value = utils::StringUtils::to_base64(
+          utils::crypto::encrypt(*property_value, property_key, encryption_key));
+
+      update(property_key, encrypted_property_value);
+
+      if (encryption_type) {
+        update(encryption_type_key, utils::crypto::EncryptionType::name());
+      } else {
+        insertAfter(property_key, encryption_type_key, utils::crypto::EncryptionType::name());
+      }
+
+      std::cout << "Encrypted property: " << property_key << '\n';
+      ++num_properties_encrypted;
+    }
+  }
+
+  return num_properties_encrypted;
+}
+
+std::vector<std::string> ConfigFile::getSensitiveProperties() const {
+  std::vector<std::string> sensitive_properties(DEFAULT_SENSITIVE_PROPERTIES.begin(), DEFAULT_SENSITIVE_PROPERTIES.end());
+  utils::optional<std::string> additional_sensitive_props_list = getValue(ADDITIONAL_SENSITIVE_PROPS_PROPERTY_NAME);
+  if (additional_sensitive_props_list) {
+    std::vector<std::string> additional_sensitive_properties = utils::StringUtils::split(*additional_sensitive_props_list, ",");
+    sensitive_properties = mergeProperties(sensitive_properties, additional_sensitive_properties);
```

Comment:
additional_sensitive_properties could be moved.

Reply:
I don't want to clutter the code with too many `std::move()`s and `&&`s which are not important semantically, only might make the the code run slightly faster.  Note that is in a command-line tool which already runs in less than 0.1 second, and will only be used a few times a year, at most.